lib/Target/OpenFhePke/OpenFhePkeEmitter.cpp

```diff
@@ -1191,60 +1193,73 @@ LogicalResult OpenFhePkeEmitter::printOperation(
 LogicalResult OpenFhePkeEmitter::printOperation(
     openfhe::MakePackedPlaintextOp op) {
   std::string inputVarName = variableNames->getNameForValue(op.getValue());
-  std::string filledPrefix =
-      variableNames->getNameForValue(op.getResult()) + "_filled";
-  std::string inputVarFilledName = filledPrefix;
-  std::string inputVarFilledLengthName = filledPrefix + "_n";
-
   FailureOr<Value> resultCC = getContextualCryptoContext(op.getOperation());
   if (failed(resultCC)) return resultCC;
   std::string cc = variableNames->getNameForValue(resultCC.value());
 
-  // cyclic repetition to mitigate openfhe zero-padding (#645)
-  os << "auto " << inputVarFilledLengthName << " = " << cc
-     << "->GetCryptoParameters()->GetElementParams()->GetRingDimension() / "
-        "2;\n";
-  os << "auto " << inputVarFilledName << " = " << inputVarName << ";\n";
-  os << inputVarFilledName << ".clear();\n";
-  os << inputVarFilledName << ".reserve(" << inputVarFilledLengthName << ");\n";
-  // inputVarFilledLengthName is unsigned
-  os << "for (unsigned i = 0; i < " << inputVarFilledLengthName << "; ++i) {\n";
-  os << "  " << inputVarFilledName << ".push_back(" << inputVarName << "[i % "
-     << inputVarName << ".size()]);\n";
-  os << "}\n";
+  if (skipVectorResizing_) {
+    emitAutoAssignPrefix(op.getResult());
+    os << cc << "->MakePackedPlaintext(" << inputVarName << ");\n";
```

Comment:
Readability nit: prefer to `return success()` in the "early exit" branch and avoid an indented block in the else case. Less indentation -> easier to read.

Same below on line 1240